server/src/db/migrations/20211102194503_machines_table.ts

```diff
@@ -0,0 +1,50 @@
+import { Knex } from "knex";
+
+
+export async function up(knex: Knex): Promise<void> {
+    
+    knex.schema.hasTable('MachineFamilies').then(function(exists) {
+        if (!exists) {
+            return knex.schema.createTable('MachineFamilies', function(t) {
+                t.increments('id');
+                t.string('name', 50);
+                t.text('description');
+                t.integer('training_module');
+                t.foreign('training_module').references('id').inTable('TrainingModule');
+              });
+        }
+    });
+
+    knex.schema.hasTable('Machines').then(function(exists) {
+        if (!exists) {
+          return knex.schema.createTable('Machines', function(t) {
+            t.increments('id');
+            t.string('name', 50);
+            t.integer('category');
+            t.foreign('category').references('id').inTable('MachineFamilies');
```

Comment:
I could be wrong here, but I think since you already added the column `category` you don't need to re-add the column using `t.foreign`. Instead just do `t.integer('category').references('id').inTable('MachineFamilies');`. I tried doing it this way initially too but I had issues with CASCADE not working

Reply:
Yep, that worked. Thanks!!